flowmachine/tests/conftest.py

```diff
@@ -62,13 +63,15 @@ def pytest_itemcollected(item):
             item._nodeid = item._nodeid.rstrip(".") + f" [{item._genid}]."
 
 
-@pytest.fixture
-def skip_datecheck(monkeypatch):
+@pytest.fixture(autouse=True)
+def skip_datecheck(request, monkeypatch):
     """Temporarily patches EventTableSubset so that it thinks any date is
     available, _without_ needing to touch the database. This shaves a little
     time off every `daily_location` creation.
     """
-    monkeypatch.setattr(EventTableSubset, "_check_dates", lambda x: True)
+    run_date_checks = request.node.get_closest_marker("check_available_dates", False)
```

Comment:
What does this line do? From the docstring it sounds like the purpose of `skip_datecheck` is to unconditionally patch `EventTableSubset`, so what is the purpose of this?

Reply:
Oh I see, this is needed for the `@pytest.mark.check_available_dates` decorator below. This should be clarified in the docstring then.